contracts/wallet-deployer.sol

```diff
@@ -21,46 +21,63 @@ pragma solidity ^0.4.25;
 import "./wallet.sol";
 import "./internals/controllable.sol";
 
+
+
 //// @title Wallet deployer with pre-caching if wallets functionality.
 contract WalletDeployer is Controllable {
 
-	event WalletCached(address wallet);
-	event WalletDeployed(address wallet, address owner);
+    event CachedWallet(address wallet);
+    event DeployedWallet(address wallet, address owner);
+    event MigratedWallet(address wallet, address owner);
 
-	mapping(address => address) public deployed;
-	address[] public cached;
+    mapping(address => address) public deployed;
+    address[] public cached;
 
-	address private ens;
-	bytes32 private controllerName;
-	bytes32 private oracleName;
-	uint private spendLimit;
+    address public ens;
+    bytes32 public controllerName;
+    bytes32 public oracleName;
+    uint public spendLimit;
 
     constructor(address _ens, bytes32 _oracleName, bytes32 _controllerName, uint _spendLimit) Controllable(_ens, _controllerName) public {
-		ens = _ens;
-		controllerName = _controllerName;
-		oracleName = _oracleName;
-		spendLimit = _spendLimit;
-	}
-
-	function deployWallet(address owner) external onlyController {
-		if (cached.length < 1) {
-			cacheWallet();
-		}
-		address walletAddress = cached[cached.length-1];
-		cached.length--;
-		Wallet(walletAddress).transferOwnership(owner);
-		deployed[owner]=walletAddress;
-		emit WalletDeployed(walletAddress, owner);
-	}
-
-	function cacheWallet() public {
-		address walletAddress = new Wallet(address(this), true, ens, oracleName, controllerName, spendLimit);
-		cached.push(walletAddress);
-		emit WalletCached(walletAddress);
-	}
-
-    function cachedContractCount() external view returns (uint) {
+        ens = _ens;
+        controllerName = _controllerName;
+        oracleName = _oracleName;
+        spendLimit = _spendLimit;
+    }
+
+    function deployWallet(address owner) external onlyController {
+        if (cached.length < 1) {
+            cacheWallet();
+    	}
+        address walletAddress = cached[cached.length-1];
+        cached.length--;
+        Wallet(walletAddress).transferOwnership(owner);
+        deployed[owner] = walletAddress;
+        emit DeployedWallet(walletAddress, owner);
+    }
+
+    function migrateWallet(address owner, uint _spendLimit, address[] _whitelistedAddresses) external onlyController {
+        if (cached.length < 1) {
+            cacheWallet();
+    	}
+        address walletAddress = cached[cached.length-1];
+        cached.length--;
+        deployed[owner] = walletAddress;
+        IWallet(walletAddress).initializeSpendLimit(_spendLimit);
```

Comment:
why not using Wallet(walletAddress)?

Reply:
Fixed! we didn't really need the interface...